app/services/BulletPointBuilder.scala

```diff
@@ -210,14 +241,35 @@ object BulletPointBuilder {
     *
     * @return - Returns a sub-sample of the input text
     */
-  def extractLeadingMatchedWords(noOfMatchedWords: Int, text: String): String = {
+  def extractLeadingMatchedWords(noOfMatchedWords: Int, texts: List[String], textsProcessed: Int, matches: List[Match], matchesProcessed: Int): String = {
+
+    val textElements: List[Match] = notSpaceRegex.findAllMatchIn(texts(textsProcessed)).toList
 
-    val matches: List[Match] = notSpaceRegex.findAllMatchIn(text).toList
+    if (matches.size == 0) {
```

Comment:
more usual to match a list with a case Nil =>

Reply:
I've rewritten the method use Scala match/case statements. I note, however, that I had to break the method into two to reduce the cyclomatic(?) complexity.